File: frontend/src/api/modules/teamActivity.ts
import { api } from '../client';

export type TeamActivityType =
  | 'member_joined'
  | 'member_left'
  | 'role_updated'
  | 'project_updated'
  | 'milestone_completed'
  | 'new_discussion'
  | 'file_uploaded';

export interface TeamActivityItem {
  id: string;
  project_id: number;
  activity_type: TeamActivityType;
  title: string;
  description?: string;
  actor_name: string;
  actor_avatar?: string;
  metadata_info?: Record<string, any>;
  created_at: string;
}

export interface TeamActivityTimelineResponse {
  project_id: number;
  items: TeamActivityItem[];
  total: number;
  page: number;
  limit: number;
  has_more: boolean;
}

export const getTeamActivityTimeline = async (
  projectId: number,
  page: number = 1,
  limit: number = 10,
  activityType?: string
): Promise<TeamActivityTimelineResponse> => {
  let url = `/projects/${projectId}/activity-timeline?page=${page}&limit=${limit}`;
  if (activityType) {
    url += `&activity_type=${activityType}`;
  }
  return await api.get<TeamActivityTimelineResponse>(url);
};

File: backend/app/schemas/team_activity.py
from __future__ import annotations

from enum import Enum
from typing import Any, Dict, List, Optional
from datetime import datetime
from pydantic import BaseModel, Field


class TeamActivityType(str, Enum):
    MEMBER_JOINED = "member_joined"
    MEMBER_LEFT = "member_left"
    ROLE_UPDATED = "role_updated"
    PROJECT_UPDATED = "project_updated"
    MILESTONE_COMPLETED = "milestone_completed"
    NEW_DISCUSSION = "new_discussion"
    FILE_UPLOADED = "file_uploaded"


class TeamActivityCreate(BaseModel):
    activity_type: TeamActivityType
    title: str = Field(..., min_length=2, max_length=255)
    description: Optional[str] = None
    actor_name: Optional[str] = None
    actor_avatar: Optional[str] = None
    metadata_info: Optional[Dict[str, Any]] = Field(default_factory=dict)


class TeamActivityItem(BaseModel):
    id: str
    project_id: int
    activity_type: TeamActivityType
    title: str
    description: Optional[str] = None
    actor_name: str
    actor_avatar: Optional[str] = None
    metadata_info: Dict[str, Any] = Field(default_factory=dict)
    created_at: datetime


class TeamActivityTimelineResponse(BaseModel):
    project_id: int
    items: List[TeamActivityItem]
    total: int
    page: int
    limit: int
    has_more: bool
